server/db/mysql/adapter.go

```diff
@@ -65,16 +74,21 @@ func (a *adapter) Open(jsonconfig json.RawMessage) error {
 		return errors.New("mysql adapter failed to parse config: " + err.Error())
 	}
 
-	a.dsn = config.DSN
-	if a.dsn == "" {
-		a.dsn = defaultDSN
+	if config.DBName == "" {
+		config.DBName = defaultDatabase
 	}
-
 	a.dbName = config.DBName
-	if a.dbName == "" {
-		a.dbName = defaultDatabase
+
+	if config.DSN != "" {
+		return errors.New(fmt.Sprintf(mysqlConfigErrorText, "dsn"))
+	}
+
+	if config.Database != "" {
+		return errors.New(fmt.Sprintf(mysqlConfigErrorText, "database"))
 	}
 
+	a.dsn = config.FormatDSN()
```

Comment:
Forgot to mention. The driver's docs say:
```
NewConfig creates a new Config and sets default values.
```
It indeed sets some default values. We cannot use the `NewConfig` so the default values are not set in our case. Please check that the DSN created from our config is not affected by these unset default values. Thanks.

Reply:
Yeah, that's a good observation. I am now using `NewConfig` to set the default values in `mysql.Config`.